llvm/lib/Target/DirectX/DXILWriter/DXILWriterPass.cpp

```diff
@@ -70,8 +116,17 @@ class EmbedDXILPass : public llvm::ModulePass {
     // Only the output bitcode need to be DXIL triple.
     M.setTargetTriple(Triple("dxil-ms-dx"));
 
+    // Perform late legalization of lifetime intrinsics that would otherwise
+    // fail the Module Verifier if performed in an earlier pass
+    legalizeLifetimeIntrinsics(M);
+
     WriteDXILToFile(M, OS);
 
+    // We no longer need lifetime intrinsics after bitcode serialization, so we
+    // simply remove them to keep the Module Verifier happy after our
+    // not-so-legal legalizations
```

Comment:
You might consider mentioning that any function containing a lifetime intrinsic gets entirely removed, rather than just the intrinsic instructions.

Reply:
It's only the lifetime intrinsic function declaration that is removed, no?